sentry_logging/lib/src/logging_integration.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:async';
+
+import 'package:logging/logging.dart';
+import 'package:sentry/sentry.dart';
+import 'extension.dart';
+
+/// An [Integration] which listens to all messages of the
+/// [logging](https://pub.dev/packages/logging) package.
+class LoggingIntegration extends Integration<SentryOptions> {
+  /// Creates the [LoggingIntegration].
+  ///
+  /// Setting [logExceptionAsEvent] to true (default) captures all
+  /// messages with errors as an [SentryEvent] instead of an [Breadcrumb].
+  /// Setting [logExceptionAsEvent] to false captures everything as
+  /// [Breadcrumb]s.
+  LoggingIntegration({bool logExceptionAsEvent = true})
+      : _logExceptionsAsEvents = logExceptionAsEvent;
+
+  final bool _logExceptionsAsEvents;
+  late StreamSubscription<LogRecord> _subscription;
+  late Hub _hub;
+
+  @override
+  FutureOr<void> call(Hub hub, SentryOptions options) {
+    _hub = hub;
+    _subscription = Logger.root.onRecord.listen(
+      _onLog,
+      onError: (Object error, StackTrace stackTrace) {
+        _hub.captureException(error, stackTrace: stackTrace);
+      },
+    );
+    options.sdk.addIntegration('LoggingIntegration');
```

Comment:
not only the integration, since this is also a new package, we should add the package to the options.sdkVersion.packages, See Timber https://github.com/getsentry/sentry-java/blob/main/sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberIntegration.kt#L44-L54